evennia/contrib/currency.py

```diff
@@ -0,0 +1,374 @@
+"""
+Currency Module
+'Currency' classes represent monetary values on objects or characters.
+They are instantiated by the 'CurrencyHandler' object, which is typically
+set up as a property on the object or character's typeclass.
+
+**Setup**
+    To use currency on an object, add a function that passes the object
+    itself into the constructor and returns a 'CurrencyHandler'. This 
+    function should be decorated with the 'lazy_property' decorator.
+
+    Example:
+    ```python
+        from evennia.utils import lazy_property
+        from world.currency import CurrencyHandler
+
+        Class Object(DefaultObject):
+        ...
+        @lazy_property
+        def purse(self):
+            return CurrencyHandler(self, db_attribute='purse')
+    ```
+**Currency Configuration**
+    'Currency' objects can be configured with a name and a value (relative to other

```

Comment:
It should be clarified that `Currency` objects are returned (and appear as properties on) the `CurrencyHandler`. It should also be made clear that `Currency` is normal Python object useful for doing computation and conversion only. As it's phrased one might think they are database entities that could be used to represent coins etc.